src/Snackbar/Snackbar.react.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable import/no-unresolved, import/extensions */
+import React, { PropTypes, PureComponent } from 'react';
+import { View, Text, Animated, Easing, Platform, StyleSheet } from 'react-native';
+
+import Button from '../Button';
+
+const propTypes = {
+    /**
+    * The text message to display.
+    */
+    message: PropTypes.string.isRequired,
+    /**
+    * Whether or not the snackbar is visible.
+    */
+    visible: PropTypes.bool.isRequired,
+    /**
+    * The amount of time in milliseconds to show the snackbar.
+    */
+    timeout: PropTypes.number.isRequired,
+    /**
+    * Callback for when the timeout finishes.
+    */
+    onRequestClose: PropTypes.func.isRequired,
+    /**
+    * Whether or not there is a bottom navigation on the screen.
+    */
+    bottomNavigation: PropTypes.bool.isRequired,
+    /**
+    * The function to execute when the action is clicked.
+    */
+    actionHandler: PropTypes.func,
+    /**
+    * The function to execute when the action is clicked.
+    */
+    actionText: PropTypes.string,
+    /**
+    * Inline style of snackbar
+    */
+    style: PropTypes.shape({
+        container: View.propTypes.style,
+        message: View.propTypes.style,
+        actionContainer: View.propTypes.style,
+        actionText: View.propTypes.style,
```

Comment:
We should prefer to reuse the button component here. So, there could be something like:

`button: Button.propTypes.style` and then you can pass it through props to button ..